src/NuGet.Core/NuGet.Protocol.Core.v3/Providers/ServiceIndexResourceV3Provider.cs

```diff
@@ -152,6 +102,92 @@ public ServiceIndexResourceV3Provider()
             return new Tuple<bool, INuGetResource>(index != null, index);
         }
 
+        // Read the source's end point to get the index json.
+        // An exception will be thrown on failure.
+        private async Task<ServiceIndexResourceV3> GetServiceIndexResourceV3(
+            SourceRepository source,
+            DateTime utcNow,
+            ILogger log,
+            CancellationToken token)
+        {
+            var url = source.PackageSource.Source;
+            var httpSourceResource = await source.GetResourceAsync<HttpSourceResource>(token);
+            var client = httpSourceResource.HttpSource;
+            
+            for (var retry = 0; retry < 3; retry++)
+            {
+                var cacheContext = HttpSourceCacheContext.CreateCacheContext(new SourceCacheContext(), retry);
```

Comment:
SourceCacheContext is disposable and disposing of it deletes all temp files (the location is static). I don't know if this is capable of creating temp files, but if another call disposes of it's context it will delete the file during a call on xplat.

That behavior may need to be changed separately.